### ISSUE TITLE
Add meta-views where all data from several projects is pooled together

### ISSUE THREAD

**Issue by WhiteNoiseDev** [marcusolsson/obsidian-projects#140]
### What would you like to be added?

I would like to have meta-views where I can see/operate on data from multiple projects.

### Why is this needed?

Since most likely you are working on multiple projects at the same time, it would be useful to have a bird view of everything together. Especially the Card View. See the backlog, doing, done for everything you are working on.

PS: Thank you very much for developing this plugin. It has the wow factor.

**Comment by WhiteNoiseDev** [comment]
Trello calls them "Workspace Views".

**Comment by marcusolsson** [comment]
I didn't know Trello had something like that! Thank you for the tip! 🙏 I'll look into it!

**Comment by puertoric** [comment]
I have the same use case. My workaround is to create a top-level folder and for each project a subfolder. I create a project for the top-level folder and activate „Include subfolders“ in the project settings. Now I have a view over all projects (which are represented by the subfolders). For the moment this is enough for my use case.

By the way, you‘re Plugin is amazing and a great deal! Thank you Marcus.

**Comment by leethobbit** [comment]
I'd like to second this, but maybe with a twist - I like the default table view, but would like to see each Projects category shown, with associated notes.  I have four Projects categories - Migrations, PS Projects, Personal, and Work - Internal Projects.  I'd love to have the ability to just show them all in separate tables down the screen, instead of selecting the dropdown to constantly switch between them. I jump between these categories a lot and going to the dropdown each time feels tedious compared to the rest of the experience.

Thanks for the amazing plugin! This plugin accomplishes the same thing I was doing with 6 other plugins and lots more maintenance on my part.

**Comment by marcusolsson** [comment]
I'm trying to figure out how this would work and would love to understand the need a little bit more.

As far as I understand, you'd like to see notes from multiple projects at the same time. What I'm trying to figure out is whether this is a change in the views, in the project definition, or whether this is a whole new thing. Would love some ideas on how this would work in practice.

For example, I could imagine this as a "derived" project in a way. Instead of configuring a path or Dataview query, you'd select other projects as the data source for the project. In other words, it would look and feel the same way as any other project, but it simply include the notes from other projects instead.

**Comment by WhiteNoiseDev** [comment]
> For example, I could imagine this as a "derived" project in a way. Instead of configuring a path or Dataview query, you'd select other projects as the data source for the project. In other words, it would look and feel the same way as any other project, but it simply include the notes from other projects instead.

I see it this way as well. It would be useful if in the relative row (in the table), or card (in board), etc, has a field showing which project it is coming from. 
As a separate FR, you can add the option to move rows/cards from one project to another.

**Comment by leethobbit** [comment]
Pictures might better explain the comments I made, so here's an Imgur (only 2 pictures!) with a description of what I was adding. You can find it here: https://imgur.com/a/42vqnjy

Realistically, this could probably be either a new view, or an enhancement of the existing views.

Thanks for your consideration!

**Comment by marcusolsson** [comment]
Thank you both for the feedback! That's indeed very helpful.

The derived project could definitely add a field to show the project it belongs to.

I wonder then if @leethobbit idea could then be implemented by a Group by setting for the table, where you'd be able to select the project field.

Also, I love the idea of moving notes between projects!

Very exciting about where this is going!

**Comment by ianscottknight** [comment]
Is this in the works? I would love this feature.

**Comment by marcusolsson** [comment]
Not in the works yet. There are a numbers of ways to implement this, so we'd need some more design first.

I have limited time to work on the plugin until the end of August, but if anyone wants to start working on some design work, I'd be happy to share feedback.

I advise not to start working on an implementation until we can agree on how it should work.

**Comment by roaldarbol** [comment]
> I have the same use case. My workaround is to create a top-level folder and for each project a subfolder. I create a project for the top-level folder and activate „Include subfolders“ in the project settings. Now I have a view over all projects (which are represented by the subfolders). For the moment this is enough for my use case.

@puertoric This will be my approach for now too I think. How do you point to the folder? I'm trying to use dataview `file.folder`, but I can't get it to work. Could you share a minimal example?

**Comment by roaldarbol** [comment]
Having looked at it more, I've come up with a reasonable workflow. @marcusolsson Would it be possible to expose `folder`(`folder_name`?) in addition to `name` and `path`? That way it is possible to create a Kanban that uses the folder name (which in this scenario is the project name), to show tasks by project.

**Comment by marcusolsson** [comment]
@roaldarbol We could add the folder name as a built-in field if that would help.

Out of curiosity, what would you expect the `folder` field to say for the following notes:

- `/Work/Projects/My note.md`
- `/Personal/Projects/My other note.md`

Would it be just `Projects`, or `/Work/Projects` and `/Personal/Projects`?

The latter would essentially be `basename`.

**Comment by roaldarbol** [comment]
In my particular case I would expect just `Projects`....maybe... actually a good question.

My current setup just has a folder called `Projects`, and inside that I have a folder for each project, e.g. `001-2Photon`,`002-Behaviour`, etc. But I was planning to have sub-folders inside that are identical between projects, so:

- `001-2Photon/Tasks`
- `002-Behaviour/Tasks`

So I guess, either I just omit the subfolder and have the tasks freely within the individual project folders, or maybe there could be a way to expose folders at different depths, maybe something that could be set under `More settings` within `Edit project`? So an option where you could choose `Expose additional fields`, where you could choose different fields, one of them being different path depths (e.g. a depth of one would be `001-2Photon` and `002-Behaviour`, whereas a depth of 2 would be `Tasks` and `Tasks`), but `basename` could also be chosen.

**Comment by roaldarbol** [comment]
Just to follow up here, I think three actions could be taken to make for a workable solution (which could then be improved with time once we've had more time to test it). 

- Expose `folder` as `Space`
- Expose `basename` as `folder`/`destination`
- In `Create new note`, create an optional field to choose where the note is placed (`basename`), and auto-fill `Space` with the first-layer folder. 

How does that sound @marcusolsson @Acylation?

**Comment by marcusolsson** [comment]
Regardless of this feature request, I think a Group by option for the views makes perfect sense. It sounds like it would be able to serve this use case and others.

Regarding your last suggestion, Projects already have a "Location for new notes", so I'm a little wary to adding another configuration that does the same thing. I'd be open to letting the user override the "Location for new notes" in the New note dialog. Would that work?

## Additional built-in fields

I propose that we add two new built-in fields called `foldername` and `folderpath`.

Example:

| name | path | foldername | folderpath | 
|-|-|-|-|
| Note 1 | `001-2Photon/Tasks/Note 1.md` | `Tasks` | `001-2Photon/Tasks` |
| Note 2 | `002-Behaviour/Tasks/Note 2.md` | `Tasks` | `002-Behaviour/Tasks` |

## Group by

### Table

For the Table view, I'm thinking something similar to how Monday does is (see below). Where each unique value becomes its own section.

![image 1 (29)](https://github.com/marcusolsson/obsidian-projects/assets/8396880/a2f9d0b9-3950-46c5-ad35-125f39e236b5)

### Board

Arguably, Board view already have a Group by feature, to group notes in columns.

### Calendar

I imagine Group by would create separate calendars. Not sure how much that would add since you can already color by field.

### Gallery

Think this would look similar to the Table view.

**Comment by marcusolsson** [comment]
> You would then specify it just for the new note you are creating right?

Correct. The difference though would be that you'd select from all folders, rather than limit to only the unique values of `Space` in your case.

**Comment by roaldarbol** [comment]
Ah, then we did misunderstand each other - I'm thinking exactly the same as you! 😃 I just meant that the new additional built-in fields would be added to the note - which they would by default.

**Comment by Juno-T** [comment]
I think the current version is already capable of 80% of this feature, although the setup is quite complicated. Here is how you can do it:

1. The idea is to have 'master' project that use tag-based tasks, let's say the tag is called `#task`. And then, for each of your projects e.g. blogging projects, learning projects etc., you add the `#task` mutual tag to every note you create.
2. Now, you can also add common columns to every project such as `_project`, `_created_at`, `_status` and these columns will show up in your master project. You can also color-code your tasks based on project or status, and use calendar view based on the created_at column.
3. Finally to make your life easier, you can create a template for each of your project that is `#task` tagged, and `_project` column filled with the project name. So that every task you create from now on will show up in the master project.

This is my master project board view:
![Image](https://github.com/user-attachments/assets/b9dbb760-b498-498e-9d8e-d8bbafd93c26)


---
### Beside the complicate setup process, there are a few things that this approach cannot address:
- **How to migrate old projects to this setup?** 
Changing the name of the status column to be a common name is easy. But adding tag to every note is not. Not sure if there is a tool to do that. I think a simple bash script will work if you know how to write one.
- **You cannot create a task from the master project** (I think this is a bug)
Well, you can, but even if you specify the project, the actual note location will be inside the master project's designated location, not your actual project's folder and will not show up in your project if you use folder-based project. I think tag-based would be fine, but still, it will not be created in each project's folder.
